### ISSUE TITLE
Nixos: Ignore packages explicitly marked as broken

### ISSUE THREAD

**Issue by nomeata** [repology/repology-updater#1340]
nixpkgs has huge coverage of the available software, and that’s partly because of auto-generated package files, e.g. for Haskell packages. But often these packaged cannot actually be built, so from a user point of view that’s almost as if they were not packaged. It’s a bit like a Debian package where there is a source tarbarll in the archive, but no binary packages could be built (and thus no binary packages would be reported to repology).

Luckily, they are usually marked as “broken”, and this information is available in the `packages.json` that is fetched by the repo updater, see the fourth line of this snippet:

```
    "adom": {
      "meta": {
        "available": false,
        "broken": true,
        "description": "A rogue-like game with nice graphical interface",
        "homepage": "http://adom.de/",
        "insecure": false,
        "license": {
          "deprecated": false,
          "free": false,
          "fullName": "Unfree redistributable",
          "redistributable": true,
          "shortName": "unfreeRedistributable"
        },
        "maintainers": [
          {
            "email": "grrwlf@gmail.com",
            "github": "grwlf",
            "githubId": 4477729,
            "name": "Sergey Mironov"
          }
        ],
        "name": "adom-1.2.0_pre23-noteye",
        "outputsToInstall": [
          "out"
        ],
        "platforms": [
          "x86_64-linux"
        ],
        "position": "pkgs/games/adom/default.nix:51",
        "unfree": true,
        "unsupported": false
      },
      "name": "adom-1.2.0_pre23-noteye",
      "outputName": "out",
      "outputs": {
        "out": null
      },
      "pname": "adom",
      "system": "x86_64-linux",
      "version": "1.2.0_pre23-noteye"
    },
```

I suggest to ignore any package marked as broken in [the repo updater](https://github.com/repology/repology-updater/blob/master/repos.d/nixos.yaml).

**Comment by nomeata** [comment]
This might also alleviate the concerns in https://github.com/repology/repology-updater/issues/638 a bit.

**Comment by AMDmi3** [comment]
I don't think we need this, or rather it should be handled differently and on a global level. Broken packages don't really change a thing for repology - unless the version is incorrect these are still valid sources of even latest versions, and for older versions ignores don't do anything anyway. Regarding 'the repository does not provide this package for users in fact' case, we don't handle this properly anyway - repology mostly tracks source packages and does not even consider binary packages if source package info is available, so except for a quite few cases where we track binary repositories we can't even tell that the package is not available for e.g. non-x86, or is available at all.

**Comment by nomeata** [comment]
At least in the case of Haskell packages, the “sources” are auto-generated from Hackage, so there are 10000 packages reported as present that nobody ever built, let alone manually touched. As a user of repology, reporting them as being part of nixpkgs is confusing and misleading in a way.

The same problem exists on hackage (which has its own distro-coverage-report), and we are fixing it there as well: https://github.com/NixOS/nixpkgs/pull/243601

**Comment by nomeata** [comment]
> rather it should be handled differently and on a global level.

That sounds good as well, of course. What do you have in mind?

**Comment by AMDmi3** [comment]
Not much in particular.
